components/automate-gateway/gateway/middleware/auth_interceptor.go

```diff
@@ -182,7 +182,7 @@ func getProjectsFromMetadata(projectHeaderEntries []string) []string {
 	for _, entry := range projectHeaderEntries {
 		for _, project := range strings.Split(entry, ",") {
 			newProject := strings.TrimSpace(project)
-			if _, value := keys[newProject]; !value {
+			if !keys[newProject] {
```

Comment:
I'm sort of surprised this worked-- I would expect that if the value of `keys[newproject]` was nil, this would fail.

Don't we need to both check the value and the success of extracting it (`if val, ok := keys[newProject]; ok {`)?

Maybe I'm missing something 🤔

Reply:
`keys[newProject]`, if not present, will return the zero-value for the type, in this case `false`. So this statement is still asserting "if there is no newProject entry in keys then..."

Reply:
@msorens I know that it reads that way, but I was under the impression that golang didn't necessarily behave how we'd expect here. I've tried this in a playground and I keep getting the error I'd expect (invalid operation)-- would you mind linking me to an example plaground?

Reply:
https://play.golang.org/p/y1GXa4D4vdv does that do the trick? Go really isn't dynamic at all, but non-existing values in maps will return their zero value, as @msorens suggested. [Here's the spec section](https://golang.org/ref/spec#Index_expressions):

>  For a of map type M:
>
> - x's type must be assignable to the key type of M
> - if the map contains an entry with key x, a[x] is the map element with key x and the type of a[x] is the element type of M
> - if the map is nil or does not contain such an entry, **a[x] is the zero value for the element type of M**

Reply:
Ugh thank you for the example-- I'd read the docs but was having trouble manually testing . . . I had my map constructed incorrectly (the value as a string, not a bool) 🙄